setup.sh

```diff
@@ -37,6 +37,13 @@ main() {
 	SRC_DIR="https://github.com/${REPOSITORY}/archive/"
 	API_URL="https://api.github.com/repos/${REPOSITORY}/commits/${BRANCH}"
 	DOC_URL="https://github.com/${REPOSITORY}/tree/${BRANCH}#installation-on-openwrt"
+	CURRENT_SETUP_SH="https://raw.githubusercontent.com/${REPOSITORY}/${BRANCH}/setup.sh"
+
+	# Get the current setup.sh script and execute it
+	if [ -z "${__CAKE_AUTORATE_URL_LOADED:-}" ]
```

Comment:
What does this do?

Reply:
When the setup.sh file is run, it fetches the branch's latest setup.sh from the web and then executes it; but it needs to figure out whether it should keep fetching from the latest or not; so `__CAKE_AUTORATE_URL_LOADED` is a marker for the script to know that it shouldn't fetch itself.

Reply:
Otherwise you'd have an infinite loop of it just constantly loading itself up from the web and executing it.